### ISSUE TITLE
Scoped modules with the same name as orgName.

### ISSUE THREAD

**Issue by RobertBrewitz** [single-spa/create-single-spa#109]
Good day! Out of the box single-spa uses a regexp to assign all modules scoped with the orgName as external modules.

https://github.com/single-spa/create-single-spa/blob/master/packages/webpack-config-single-spa/lib/webpack-config-single-spa.js#L72

## Problem encountered

What we tried to do before realizing this, was to import npm modules with the same scope name as the orgName.

```javascript
// webpack.config.js
  const defaultConfig = singleSpaDefaults({ orgName: 'my-awesome-org' })

// root.component.tsx
  import * as MyAwesomeOrgPackage from '@my-awesome-org/package';
```

Out of the box this didn't work and webpack treated `@my-awesome-org/package` as an external dependency.

## Hacky solutions

Whilst writing this ticket up we came up with two separate workarounds.

One workaround is to alias `@my-awesome-org/package` without the scope name i.e. `package`.

E.g.

```javascript
// root.component.tsx
  import * as MyAwesomeOrgPackage from 'package';

// tsconfig.json
  "paths": {
    "package": ["@my-awesome-org/package"]
  }

// webpack.config.json
alias: {
  "package": "@my-awesome-org/package",
}
```
Another workaround the we are currently running with is to omit the regexp from the singleSpaDefaults for the specific apps that require the package.

E.g.

```javascript
const ORG_NAME = "my-awesome-org";

module.exports = (webpackConfigEnv) => {
  const defaultConfig = singleSpaDefaults({
    orgName: ORG_NAME,
    projectName: "header",
    webpackConfigEnv,
  });

  defaultConfig.externals = defaultConfig.externals.filter(
    (lib) => !new RegExp(lib).toString().includes(`@${ORG_NAME}`)
  );

  return webpackMerge.smart(defaultConfig, {
    module: {
      rules: [
        {
          test: /\.(s*)css$/,
          use: ["style-loader", "css-loader", "sass-loader"],
        },
      ],
    },
    resolve: {
      extensions: [".js", ".jsx", ".scss"],
    },
  });
};
```

## Request

I am not sure this is an issue really, might just be nice to have some documentation on this topic :)

Kind regards,
Robert

**Comment by joeldenning** [comment]
Hi @RobertBrewitz, good question. We mark things under the `@org` npm scope as externals to encourage [in-browser modules](https://single-spa.js.org/docs/recommended-setup#in-browser-versus-build-time-modules), which are a really powerful way of sharing code between microfrontends. To use in-browser modules, you'd add the scoped packages to your import map so that they are shared as deployable, in-browser modules.

With all that in mind, you are absolutely right that default webpack config does not allow using *any* scoped packages under your organization as build-time dependencies instead of as in-browser dependencies. The changes to the webpack config that you've shown above are what I'd recommend for disabling this behavior.

Regarding documentation, I'd gladly accept a PR updating the documentation. Here's the file that would be updated - https://github.com/single-spa/single-spa.js.org/edit/master/website/versioned_docs/version-5.x/create-single-spa.md

Additionally, I'm open to considering further configuration options to webpack-config-single-spa that support either fully or partially disabling this behavior. Many organizations have a combination of in-browser and build-time modules, and so they might want some things to be webpack externals and other things not to be externals. I'd be open to adding configuration options to support that.

**Comment by filoxo** [comment]
https://github.com/single-spa/create-single-spa/pull/199 adds the ability to disable the org scope through the webpack config released in [v1.16.0](https://github.com/single-spa/create-single-spa/releases/tag/v1.16.0).
